### ISSUE TITLE
SocketsForCordova: Cannot register WP event dispatcher, Error: Missing Command Error

### ISSUE THREAD

**Issue by twy30** [OpenPTT/OpenPTT#15]
Environment: WinPro8.1 + VS2013 + WP81 emulator

Repro: Deploy app to WP81 emulator

Deployment to WP81 emulator was successful; however upon app init JavaScript Console says `SocketsForCordova: Cannot register WP event dispatcher, Error: Missing Command Error`.  Login doesn't seem to work, and this is probably why.

`SocketsForCordova` does state that it supports WP8, but it's not clear whether WP81 is supported.

I can try debugging this but given that I am new to Cordova, this will take me a while.

**Comment by ettoolong** [comment]
建議先建一個最簡單的 cordova sample, 然後裝一下該 plugin, 寫幾行 code 測看看正不正常.
我看 plugin 頁是寫有支援 WP 的.

**Comment by twy30** [comment]
will do.

**Comment by twy30** [comment]
I was able to repro this with following steps:

## HelleWorld Project Setup

```
cordova create hello com.example.hello HelloWorld
cd .\hello
cordova platform add windows
cordova plugin add cz.blocshop.socketsforcordova
```

## Test Code

```
onDeviceReady: function() {
    app.receivedEvent('deviceready');
    var socket = new Socket();
    socket.onData = function (data) {
        console.log(data);
    };
    socket.open(
        "google.com",
        80,
        function () { console.log("socket.open() succeeded"); },
        function (errorMessage) { console.log("socket.open() failed"); console.log("socket: " + errorMessage); });


},
```

## On WP81 Emulator

Project: `hello\platforms\windows\CordovaApp.Phone.jsproj`

JavaScript Console:

```
Received Event: deviceready
socket.open() failed
socket: Missing Command Error
SocketsForCordova: Cannot register WP event dispatcher, Error: Missing Command Error
```

## On W81 Simulator

Project: `hello\platforms\windows\CordovaApp.Windows.jsproj`

JavaScript Console:

```
Received Event: deviceready
socket.open() failed
socket: Missing Command Error
```

**Comment by twy30** [comment]
I also checked out http://www.blocshop.cz/blog/?p=6 and googled for `cordova missing command error` but didn't see anything relevant.

`missing command error` seems to be a generic error for cordova when plugins are not installed properly.  However I don't see how that could be cause of this (there was no errors / warnings during my HelloWorld project setup steps listed above.

I will continue digging.

**Comment by twy30** [comment]
hmm... on http://www.blocshop.cz/blog/?p=6 it listed a sample project: https://github.com/blocshop/sockets-for-cordova-tester

I am going to try that next.

**Comment by twy30** [comment]
This is probably where the problem is: https://github.com/blocshop/sockets-for-cordova/blob/master/plugin.xml .  Although `wp8` is supported, the `windows` platform (which includes W8, W8.1, W10, WP8.1) is not listed.

Let me see if I can hack it...

**Comment by twy30** [comment]
Nope... forked it and yolo copied `<platform name="wp8">` to `<platform name="windows">`; removed the original plugin, added my own -- no luck. :(

My forked repo: https://github.com/twy30/sockets-for-cordova

**Comment by twy30** [comment]
I guess I can look into adding cordova socket support for the `windows` platform ( http://cordova.apache.org/docs/en/edge/guide_platforms_win8_plugin.md.html#Windows%20Plugins )  If it's worth it...

**Comment by twy30** [comment]
According to http://plugins.cordova.io/#/search?search=socket , it looks like this plugin http://plugins.cordova.io/#/package/com.tlantic.plugins.socket supports `windows8`.  I am going to try it out tonight.  Then we can decide whether we should use com.tlantic.plugins.socket or fix up cz.blocshop.socketsforcordova

**Comment by ettoolong** [comment]
在 https://github.com/blocshop/sockets-for-cordova 這專案裡面的 plugin.xml
裡加入 windows8 的設定, 然後加一個資料夾, 把 sockets-for-cordova/src/wp8/src/ 裡的東西 copy 進去.
不知道有沒有用, 感覺它是用 C# 實作了 socket 相關的部份, 只要 .NET 元件還在, 感覺應該可以用?
或是直接把:
https://github.com/Tlantic/cdv-socket-plugin 的 plugin.xml 設定和實作部份搬進我們現在用的 sockets-for-cordova 裡面看看?

有人有環境能試試嗎?
